app/scripts/views/sign_up.js

```diff
@@ -349,6 +360,10 @@ define(function (require, exports, module) {
       return /@firefox(\.com)?$/.test(email);
     },
 
+    _isPasswordNotSameAsConfirmPassword() {
```

Comment:
I find the `Not` in the middle of the name easy to miss.  This could be named `_doPasswordsMatch()` and use a `!` whenever `this._isPasswordNotSameAsConfirmPassword()` is called - `if (! this._doPasswordsMatch())`

As an aside, my suggestions are so that we can extract the common logic into a mixin for re-use with the [email-first PR](https://github.com/mozilla/fxa-content-server/pull/5177) .